src/Stats.h

```diff
@@ -0,0 +1,27 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#pragma once
+
+namespace reactivesocket {
+// forward decl for NoopStats
+class NoopStats;
+
+// Virtual interface for a stats receiver
+class Stats {
+ public:
+  static NoopStats noop;
```

Comment:
I'll try your first suggestion, I wasn't sure about singleton as it seemed to imply another folly dep.